### PR TITLE
Fix/5189 log issues

### PR DESCRIPTION
**Pull request by whiskey**
## Description
This PR fixes some issues that appeared while trying to fix the 'empty all' filter in EXPOSUREAPP-5189. 

Fixed issues:
- logs are zipped before sharing them to reduce UI lag and file size to transfer
- logs are extended to include module, line and function of the calling `Log` statement
- the log view is scrolled to the bottom (this is not fully working on very long log files; a future refactoring will tackle this once and for all)

## Link to Jira
[EXPOSUREAPP-5189](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5189)

## Screenshots
Screen Shot 2021-02-23 at 22 17 29
Screen Shot 2021-02-23 at 22 18 27
Screen Shot 2021-02-23 at 22 25 25

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/Logging/DMLogsViewController.swift [hunk 1]
```diff
@@ -138,7 +139,17 @@ final class DMLogsViewController: UIViewController {
...
-		let activityViewController = UIActivityViewController(activityItems: [logString], applicationActivities: nil)
+
+		guard
+			let item = LogDataItem(logString: logString)
+		else {
+			Log.warning("No log data to export.", log: .localData)
+			return
+		}
+
+		// Log.debug("Preparing log export. Log length, raw: \(rawData.count), compressed: \(archive.data?.count ?? 0)", log: .localData)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Developer Menu/Features/Logging/DMLogsViewController.swift:150]: Should we remove this?

**Review by naveeddotio — APPROVED** [review]

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Workers/SQLiteKeyValueStore.swift [hunk 2]
```diff
@@ -248,9 +248,9 @@ final class SQLiteKeyValueStore {
 					return value
 				}
 			} catch let DecodingError.typeMismatch(expectedType, context) {
-				Log.warning("Type mismatch in K/V decoding: \(expectedType) not found in \(context). Trying fallback implementation.", log: .localData)
+				Log.warning("Type mismatch for key \(key) in K/V decoding: \(expectedType) not found in \(context). Trying fallback implementation.", log: .localData)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Workers/SQLiteKeyValueStore.swift:251]: 👌

File: src/xcode/ENA/ENA/Source/Workers/SQLiteKeyValueStore.swift [hunk 3]
```diff
@@ -248,9 +248,9 @@ final class SQLiteKeyValueStore {
...
 			} catch {
-				Log.error("Error when decoding value from K/V SQLite store: \(error.localizedDescription)", log: .localData, error: error)
+				Log.error("Error when decoding value for key \(key) from K/V SQLite store: \(error.localizedDescription)", log: .localData, error: error)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Workers/SQLiteKeyValueStore.swift:253]: 💎

File: src/xcode/ENA/ENA/Source/Workers/SQLiteKeyValueStore.swift [hunk 4]
```diff
@@ -259,7 +259,7 @@ final class SQLiteKeyValueStore {
 				let encoded = try JSONEncoder().encode([newValue])
 				try setData(encoded, for: key)
 			} catch {
-				Log.error("Error when encoding value for inserting into K/V SQLite store: \(error.localizedDescription)", log: .localData)
+				Log.error("Error when encoding value for key \(key) inserting into K/V SQLite store: \(error.localizedDescription)", log: .localData)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Workers/SQLiteKeyValueStore.swift:262]: 👏

**Review by pascal-brause — APPROVED** [review]

**Pull request merged**